src/components/Header/Header.component.jsx

```diff
@@ -1,33 +1,137 @@
 import React, { useContext } from 'react';
-import { BiMenu } from 'react-icons/bi';
-import GeneralContext from '../../state/GeneralContext';
-import ThemeContext from '../../state/ThemeContext';
+import { Link } from 'react-router-dom';
+import { styled, useTheme } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import CssBaseline from '@mui/material/CssBaseline';
+import List from '@mui/material/List';
+import Typography from '@mui/material/Typography';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import { BiChevronLeft, BiChevronRight, BiMenu } from 'react-icons/bi';
+import ListItem from '@mui/material/ListItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import SearchBar from './SearchBar.component';
+import ToggleSwitch from './ToggleSwitch.components';
 import {
   StyledHeader,
-  StyledRowContainer,
   StyledUserIcon,
+  StyledRowContainer,
 } from '../../styled/Header.styled';
-import ToggleSwitch from './ToggleSwitch.components';
-import SearchBar from './SearchBar.component';
+import GeneralContext from '../../state/GeneralContext';
+import ThemeContext from '../../state/ThemeContext';
+
+const drawerWidth = 240;
+
+const Main = styled('main', { shouldForwardProp: (prop) => prop !== 'open' })(
+  ({ theme, open }) => ({
+    flexGrow: 1,
+    padding: theme.spacing(3),
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+    marginLeft: `-${drawerWidth}px`,
+    ...(open && {
+      transition: theme.transitions.create('margin', {
+        easing: theme.transitions.easing.easeOut,
+        duration: theme.transitions.duration.enteringScreen,
+      }),
+      marginLeft: 0,
+    }),
+  })
+);
```

Comment:
we usually create a separate file with all the styled components. This could help us maintaining a clean and organized code.